### PR TITLE
Fea 247 task 276

### PR DESCRIPTION
**Pull request by kchoadley**

### CODE REVIEW

File: rest-server/CMakeLists.txt [hunk 1]
```diff
@@ -54,6 +54,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++17 -DBOOST_LOG_DYN_LINK")
 # source files
 add_executable(${PROJECT_NAME}  
                                 ./src/rest_server.cpp
+                                ./src/controller/disseminator_controller.cpp
```
Comment (kchoadley) [inline rest-server/CMakeLists.txt]: File names not in alphabetical order

File: rest-server/src/controller/disseminator_controller.hpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+     */
+    void HandlePut(http_request message) override;
+    /// Creates a new disseminator.
+    /**
+     * HandlePost will add a new disseminator or collection of disseminators to this endpoint.
+     * The body should include json of the disseminators to be posted.
+     * 
+     * @param message The http request message to be parsed.
+     */
+    void HandlePost(http_request message) override;
```
Comment (vsagrawal0) [inline rest-server/src/controller/disseminator_controller.hpp]: For documentation, I think this function should specify what a proper body for the POST would look like.

File: cadg-db/create_cadg_db.sql [hunk 3]
```diff
@@ -55,12 +55,13 @@ COMMENT = '				';
 DROP TABLE IF EXISTS `cadg`.`disseminator` ;
 
 CREATE TABLE IF NOT EXISTS `cadg`.`disseminator` (
```
Comment (vsagrawal0) [inline cadg-db/create_cadg_db.sql]: Based on our conversations with Reza, I was under the impression that every disseminator should have an active IP and a standby IP, along with the CMAC version being used for each. Is this being considered in this db design?

File: rest-server/src/controller/disseminator_controller.hpp (hunk 2 above)
Comment (vsagrawal0) [inline rest-server/src/controller/disseminator_controller.hpp]: For documentation, this function should specify what a proper POST body contains.

Comment (kchoadley) [reply rest-server/src/controller/disseminator_controller.hpp]: My documentation was incomplete, thanks for pointing this out!

File: cadg-db/create_cadg_db.sql (hunk 3 above)
Comment (kchoadley) [reply cadg-db/create_cadg_db.sql]: I am going to wait until I know for sure everything else that we need in the db schema before I make further modifications. I have some questions. Like, does this mean we need 2 ports for each IP, or can we assume they each use the same 2 ports?

**Comment by kchoadley** [comment]
Doxygen is now up to date, and the POST example Vaniya showed is now syntactically correct.

**Comment by kchoadley** [comment]
The rest of the implementation of the Disseminator Controller is functional now.

**Pull request merged**